server/src/main/java/org/candlepin/util/Util.java

```diff
@@ -547,4 +558,37 @@ public static String getHostname() {
     public static OffsetDateTime toDateTime(Date date) {
         return date != null ? date.toInstant().atOffset(ZoneOffset.UTC) : null;
     }
+
+    /**
+     * Takes a map and splits it into a list of maps each with one item from the given map.
+     *
+     * @param map a map to be split
+     * @return a list of single item maps
+     */
+    public static List<Map<String, String>> split(Map<String, String> map) {
+        if (map == null) {
+            return Collections.emptyList();
+        }
+        return map.entrySet().stream()
+            .map(Util::singletonMap)
+            .collect(Collectors.toList());
+    }
+
+    private static Map<String, String> singletonMap(Map.Entry<String, String> entry) {
+        return singletonMap(entry.getKey(), entry.getValue());
+    }
+
+    /**
+     * Takes a key value pair and returns a map with a single entry.
+     *
+     * @param key a key of the single entry
+     * @param value a value of the single entry
+     * @return a map with a single entry
+     */
+    public static Map<String, String> singletonMap(String key, String value) {
```

Comment:
This method is already defined in ```java.util.Collections``` with the exact same name and arguments, but is generically typed.

Reply:
Done.